pysr/regressor_sequence.py

```diff
@@ -0,0 +1,189 @@
+
+    def _variable_names(self, y, variable_names=None):
+        if not variable_names:
+            if y.shape[1] == 1:
+                return [f"xt_{i}" for i in range(self.recursive_history_length, 0, -1)]
+            else:
+                return [
+                    f"x{i}t_{j}"
+                    for j in range(self.recursive_history_length, 0, -1)
+                    for i in range(y.shape[1])
+                ]
+        else:
+            return [
+                i + "t_" + str(j)
+                for j in range(self.recursive_history_length, 0, -1)
+                for i in variable_names
+            ]
+
+    def fit(
+        self,
+        X,
+        y=None,
+        Xresampled=None,
+        weights=None,
+        variable_names: Optional[ArrayLike[str]] = None,
+        complexity_of_variables: Optional[
+            Union[int, float, List[Union[int, float]]]
+        ] = None,
+        X_units: Optional[ArrayLike[str]] = None,
+        y_units=None,
+    ) -> "PySRSequenceRegressor":
+        """
+        Search for equations to fit the time series dataset and store them in `self.equations_`.
+
+        Parameters
+        ----------
+        X : ndarray | pandas.DataFrame
+            Training time series data of shape (n_times, n_features).
+        weights : ndarray | pandas.DataFrame
+            Weight array of the same shape as `X`.
+            Each element is how to weight the mean-square-error loss
+            for that particular element of `X`. Alternatively,
+            if a custom `loss` was set, it will can be used
+            in arbitrary ways.
+        variable_names : list[str]
+            A list of names for the variables, rather than "x0t_1", "x1t_2", etc.
+            If `X` is a pandas dataframe, the column name will be used
+            instead of `variable_names`. Cannot contain spaces or special
+            characters. Avoid variable names which are also
+            function names in `sympy`, such as "N".
+            The number of variable names must be equal to (n_features,).
+        X_units : list[str]
+            A list of units for each variable in `X`. Each unit should be
+            a string representing a Julia expression. See DynamicQuantities.jl
+            https://symbolicml.org/DynamicQuantities.jl/dev/units/ for more
+            information.
+            Length should be equal to n_features.
+
+        Returns
+        -------
+        self : object
+            Fitted estimator.
+        """
+
+        _check_assertions(
+            X,
+            y,
+            Xresampled,
+            self.recursive_history_length,
+            weights,
+            variable_names,
+            X_units,
+            y_units,
+        )
+
+        y = X[self.recursive_history_length :]
+        newX = np.lib.stride_tricks.sliding_window_view(
+            X[:-1].flatten(), self.recursive_history_length * X.shape[1]
+        )[:: y.shape[1], :]
```

Comment:
```suggestion
        current_X = X[self.recursive_history_length :]
        historical_X = np.lib.stride_tricks.sliding_window_view(
            X[:-1].flatten(), self.recursive_history_length * X.shape[1]
        )[:: current_X.shape[1], :]
```

I think these are more suitable variable names. Wdyt?

Reply:
yeah I think its good (I need to get better at naming variables) but aren't they supposed to be the other way around?
bcuz the stride tricks function is significantly changing the original X, so the output should be current_X

Reply:
> so the output should be current_X

I think that's what's suggested here, no?